pkg/controller/container-runtime-config/helpers_test.go

```diff
@@ -1245,22 +1264,143 @@ func TestGetValidScopePolicies(t *testing.T) {
+		},
+		{
+			name:                 "cip scope a.com/a1/a2 is super scope of registries sources",
+			clusterScopePolicies: clusterScopePolicies,
+			idmsRules: []*apicfgv1.ImageDigestMirrorSet{
+				{
+					Spec: apicfgv1.ImageDigestMirrorSetSpec{
+						ImageDigestMirrors: []apicfgv1.ImageDigestMirrors{
+							{Source: "a.com/a1/a2", Mirrors: []apicfgv1.ImageMirror{"a-a1-a2.mirror/a1/a2"}},
+							{Source: "a.com/a1/a2/a3", Mirrors: []apicfgv1.ImageMirror{"a-a1-a2-a3.mirror/m1/m2/m3"}},
+							{Source: "a.com/a1/a2/a3-1", Mirrors: []apicfgv1.ImageMirror{"a-a1-a2-a3-1.mirror/m1/m2/m3"}},
+							{Source: "x.com/x1/x2", Mirrors: []apicfgv1.ImageMirror{"x-x1-x2.mirror/m1/m2"}},
+						},
+					},
+				},
+			},
+			expectedSigstoreRegistriesConfigScopes: []string{"*.example.com", "a-a1-a2-a3-1.mirror/m1/m2/m3", "a-a1-a2-a3.mirror/m1/m2/m3", "a-a1-a2.mirror/a1/a2", "a.com/a1/a2", "a.com/a1/a2/a3", "a.com/a1/a2/a3-1", "a.com/a1/a2@sha256:0000000000000000000000000000000000000000000000000000000000000000", "policy.scope"},
+		},
+		{
+			name:                 "cip scope a.com/a1/a2 nested in registries sources",
+			clusterScopePolicies: clusterScopePolicies,
+			idmsRules: []*apicfgv1.ImageDigestMirrorSet{
+				{
+					Spec: apicfgv1.ImageDigestMirrorSetSpec{
+						ImageDigestMirrors: []apicfgv1.ImageDigestMirrors{
+							{Source: "a.com", Mirrors: []apicfgv1.ImageMirror{"a.mirror"}},
+							{Source: "a.com/a1", Mirrors: []apicfgv1.ImageMirror{"a-a1.mirror/a1"}},
+							{Source: "x.com/x1/x2", Mirrors: []apicfgv1.ImageMirror{"x-x1-x2.mirror/x1/x2"}},
+						},
+					},
+				},
+			},
+			expectedSigstoreRegistriesConfigScopes: []string{"*.example.com", "a-a1.mirror/a1/a2", "a.com/a1/a2", "a.com/a1/a2@sha256:0000000000000000000000000000000000000000000000000000000000000000", "policy.scope"},
+		},
+		{
+			name:                 "cip wildcard scope *.example.com is the mirror source",
+			clusterScopePolicies: clusterScopePolicies,
+			idmsRules: []*apicfgv1.ImageDigestMirrorSet{
+				{
+					Spec: apicfgv1.ImageDigestMirrorSetSpec{
+						ImageDigestMirrors: []apicfgv1.ImageDigestMirrors{
+							{Source: "*.example.com", Mirrors: []apicfgv1.ImageMirror{"matched.x.example.mirror"}},
+						},
+					},
+				},
+			},
+			expectedSigstoreRegistriesConfigScopes: []string{"*.example.com", "a.com/a1/a2", "a.com/a1/a2@sha256:0000000000000000000000000000000000000000000000000000000000000000", "policy.scope", "matched.x.example.mirror"},
+		},
+		{
+			name:                 "cip scope *.example.com and mirror source have wildcard matching",
+			clusterScopePolicies: clusterScopePolicies,
+			idmsRules: []*apicfgv1.ImageDigestMirrorSet{
+				{
+					Spec: apicfgv1.ImageDigestMirrorSetSpec{
+						ImageDigestMirrors: []apicfgv1.ImageDigestMirrors{
+							{Source: "a.example.com", Mirrors: []apicfgv1.ImageMirror{"a-example.mirror"}},
+							{Source: "*.x.example.com", Mirrors: []apicfgv1.ImageMirror{"matched.example.mirror", "star-x.example.mirror"}},
```

Comment:
@mtrmac PTAL. Test cases for https://github.com/openshift/machine-config-operator/pull/4449#discussion_r1671294663

Reply:
Thanks.

Also a test for policy scope `foo.example.com/ns/repo` vs `*.example.com` -> `our.mirror/example` (expected scopes `foo.example.com/ns/repo. our.mirror/example/ns/repo`) might be a interesting because it might be closer to real-world configurations (with fairly coarse mirror maps but fairly small policy scopes); I don’t expect it would change code coverage…

Reply:
We got `foo.example.com/ns/repo`, `our.mirror/example/ns/repo`, `our.mirror/example`
https://github.com/openshift/machine-config-operator/pull/4449/files#diff-e75cbabf033d7cdc22ac1877e6e0134549c848c7226729274f806c3ab943f37eR1366

Reply:
Thanks!